js/bitcoincoid.js

```diff
@@ -256,16 +255,16 @@ module.exports = class bitcoincoid extends Exchange {
     }
 
     async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
-        if (!symbol)
-            throw new ExchangeError (this.id + ' fetchOpenOrders requires a symbol');
         await this.loadMarkets ();
-        let market = this.market (symbol);
-        let request = {
-            'pair': market['id'],
-        };
+        let market = symbol ? this.market (symbol) : undefined;
+        let request = market ? { 'pair': market['id'] } : {};
         let response = await this.privatePostOpenOrders (this.extend (request, params));
-        let orders = this.parseOrders (response['return']['orders'], market, since, limit);
-        return this.filterOrdersBySymbol (orders, symbol);
+        // { success: 1, return: { orders: null }}
+        let raw = response['return']['orders'];
+        if (!raw)
+            return [];
+        let orders = this.parseOrders (raw, market, since, limit);
+        return market ? this.filterOrdersBySymbol (orders, symbol) : orders;
```

Comment:
And this one, basically it's overly sensitive to ternary conditionals, can only recognize the most simple ones...
```(market) ? noBracketsA : noBracketsB```

Reply:
This one I removed. `filtersOrdersBySymol` is robust against `undefined` symbol.